server/service/src/purchase_order_line/update/validate.rs

```diff
@@ -0,0 +1,20 @@
+use repository::{PurchaseOrderLineRow, PurchaseOrderLineRowRepository, StorageConnection};
+
+use crate::purchase_order_line::update::{
+    UpdatePurchaseOrderLineInput, UpdatePurchaseOrderLineInputError,
+};
+
+pub fn validate(
+    input: &UpdatePurchaseOrderLineInput,
+    connection: &StorageConnection,
+) -> Result<PurchaseOrderLineRow, UpdatePurchaseOrderLineInputError> {
+    let purchase_order_line =
+        PurchaseOrderLineRowRepository::new(connection).find_one_by_id(&input.id)?;
+
+    let purchase_order_line = match purchase_order_line {
+        Some(purchase_order_line) => purchase_order_line,
+        None => return Err(UpdatePurchaseOrderLineInputError::PurchaseOrderLineNotFound),
+    };
+
```

Comment:
I am guessing there would be in here some check on the PO status?

I guess we don't know what those would be and/or what the logic would look like, but I wonder if worthwhile adding a TODO flag like we have for permissions for when they're merged

Reply:
At first I was confused as to why the line would need to check the PO status. But makes sense - do we mean like we don't want to be able to update a line if status is already confirmed?

Reply:
Yeah! in OG confirmed and new POs can have lines added whereas finalised POs cannot

Reply:
Good pick up @fergie-nz. Will update PR later